utils/utils.go

```diff
@@ -432,3 +437,57 @@ func techArrayToString(techsArray []coreutils.Technology, separator string) (res
 	}
 	return strings.Join(techString, separator)
 }
+
+type UrlAccessChecker struct {
+	connected *bool
+	waitGroup *errgroup.Group
+	url       string
+}
+
+// CheckConnection checks if the url is accessible in a separate goroutine not to block the main thread
+func CheckConnection(url string) *UrlAccessChecker {
+	checker := &UrlAccessChecker{
+		url:       url,
+		waitGroup: new(errgroup.Group),
+	}
+	checker.waitGroup.Go(func() (err error) {
+		checker.connected = clientutils.Pointer(IsUrlAccessible(url))
+		return
+	})
+	return checker
+}
+
+// Checks if the url is accessible, can block until the connection check goroutine is done
+func (ic *UrlAccessChecker) IsConnected() bool {
+	if ic.connected != nil {
+		return *ic.connected
+	}
+	// wait for connection routine to finish
+	if err := ic.waitGroup.Wait(); err != nil {
+		return false
+	}
+	return *ic.connected
+}
+
+// Checks if the url is accessible
+func IsUrlAccessible(url string) bool {
+	// Build client
+	client, err := httpclient.ClientBuilder().Build()
+	if err != nil {
+		log.Debug(fmt.Sprintf("can't check access to '%s', build client:\n%s", url, err.Error()))
```

Comment:
Let's capitalize all logs in this function
```suggestion
		log.Debug(fmt.Sprintf("Can't check access to '%s', build client:\n%s", url, err.Error()))
```